csharp/ql/lib/semmle/code/csharp/frameworks/Razor.qll

```diff
@@ -0,0 +1,181 @@
+/** Provides definitions and flow steps related to Razor pages. */
+
+import csharp
+private import codeql.util.Unit
+private import codeql.util.FilePath
+private import semmle.code.csharp.frameworks.microsoft.AspNetCore
+
+/** A call to the `View` method */
+private class ViewCall extends MethodCall {
+  ViewCall() { this.getTarget().hasQualifiedName("Microsoft.AspNetCore.Mvc", "Controller", "View") }
+
+  /** Gets the `name` argument to this call, if any. */
+  string getNameArgument() {
+    exists(StringLiteral lit, int i | i in [0 .. 1] |
+      this.getTarget().getParameter(i).getType() instanceof StringType and
+      DataFlow::localExprFlow(lit, this.getArgument(i)) and
+      result = lit.getValue()
+    )
+  }
+
+  /** Gets the `model` argument to this call, if any. */
+  Expr getModelArgument() {
+    exists(int i | i in [0 .. 1] |
+      this.getTarget().getParameter(i).getType() instanceof ObjectType and
+      result = this.getArgument(i)
+    )
+  }
+
+  /** Gets the MVC action method that this call is made from, if any. */
+  Method getActionMethod() {
+    result = this.getEnclosingCallable() and
+    result = this.getController().getAnActionMethod()
+  }
+
+  /**
+   * Gets the action name that this call refers to, if any.
+   * This is either the name argument, or the name of the action method calling this if there is no name argument.
+   */
+  string getActionName() {
```

Comment:
This string appears to reflect the "order" in "View Discovery".
There is also some mention of "Views" and "Views/Shared" which is also handled below. The documentation for view discovery doesn't mention anything about "Area": Do you know, if there exist some documentation for this as well?

Reply:
Areas are documented here: https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/areas?view=aspnetcore-7.0